packages/bee-q/src/components/dropdown/bq-dropdown.tsx

```diff
@@ -0,0 +1,160 @@
+  @Prop({ reflect: true }) skidding?: number = 0;
+
+  /** Defines the strategy to position the panel */
+  @Prop({ reflect: true }) strategy?: 'fixed' | 'absolute' = 'fixed';
+
+  // Prop lifecycle events
+  // =======================
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  /**
+   * Handler to be called to check if the `bq-panel` switches state (visible/hidden).
+   * @returns CustomEvent - with value `{ opened: boolean }`
+   */
+  @Event() bqPanelChange: EventEmitter<{ opened: boolean }>;
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  // Listeners
+  // ==============
+
+  /** Listens for the 'click' event on the document object
+   * and closes the dropdown panel if the click is outside the component.
+   */
+  @Listen('click', { target: 'document', passive: true })
+  onClickOutside(event: MouseEvent) {
+    if (!this.open) return;
+
+    // Close when clicking outside of the close element
+    const path = event.composedPath();
+    if (!path.includes(this.el)) {
+      this.open = false;
+    }
+  }
+
+  /**
+   * Listens for the 'keyup' event on the window object
+   * and closes the dropdown panel if the 'Escape' key or 'Tab' key outside the component is pressed.
+   */
+  @Listen('keyup', { target: 'window', passive: true })
+  onEscape(event: KeyboardEvent) {
+    if (!this.open) return;
+
+    if (event.key === 'Escape' || (event.key === 'Tab' && !event.composedPath().includes(this.el))) {
+      this.open = false;
+    }
+  }
+
+  @Listen('bqSelect', { passive: true })
+  onItemSelect() {
+    this.open = false;
+  }
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private togglePanel = (): void => {
+    this.open = !this.open;
+  };
+
+  // render() function
+  // Always the last one in the class.
+  // ===================================
+
+  render() {
+    return (
+      <div class="bq-dropdown" part="base">
+        {/* TRIGGER ELEMENT */}
+        <div
+          class="bq-dropdown__trigger block"
+          ref={(el) => (this.trigger = el)}
+          onClick={this.togglePanel}
+          aria-haspopup="true"
+          aria-expanded={this.open ? 'true' : 'false'}
+          part="trigger"
+        >
+          <slot name="trigger" />
+        </div>
+        {/* PANEL */}
+        <bq-panel
+          class="bq-dropdown__panel"
+          distance={this.distance}
+          placement={this.placement}
+          open={this.open}
```

Comment:
We could add a panelHeight property that allows user to override the dropdown panel height. Something like:

```js
  /** When set, it will override the height of the dropdown panel */
  @Prop({ reflect: true }) panelHeight?: string;

  render() {
    const style = {
      ...(this.panelHeight && { '--bq-panel--height': this.panelHeight }),
    };

    return (
      <div class="bq-dropdown" part="base">
        ...
        {/* PANEL */}
        <bq-panel
          style={style}
          class="bq-dropdown__panel"
```